src/ngraph/frontend/onnx_import/op/matmul.cpp

```diff
@@ -97,11 +114,20 @@ namespace ngraph
                     if (left_shape.size() > 3)
                     {
                         left = reshape::collapse(left, 0, left_shape.size() - 3);
+                    }
+                    if (right_shape.size() > 3)
+                    {
                         right = reshape::collapse(right, 0, right_shape.size() - 3);
                     }
 
                     // Perform multiple small dot products
                     std::size_t groups = left->get_shape().at(0);
+                    // If we haven't make broadcasting earlier this means that one of inputs is vector,
```

Comment:
Please reword:
`If we haven't broadcast earlier this means that one of the inputs is a vector,`